src/components/MovieModal/index.tsx

```diff
@@ -38,54 +37,43 @@ function MovieModal({
   const [formData, setFormData] = useState<MovieDTO<typeof mode>>();
```

Comment:
Consider move `formData` and `mode` to Redux so that MovieModal doesn't need to take care of these data logic